source/obstacle/Obstacle.hx

```diff
@@ -1,57 +1,78 @@
 package obstacle;
 
+import flixel.FlxBasic;
 import flixel.FlxG;
 import flixel.FlxSprite;
 
-class Obstacle extends FlxSprite
-{
-	public function new(?parameters:ObstacleParameters)
-	{
+class Obstacle extends FlxSprite {
+	public function new(?parameters:ObstacleParameters) {
 		super();
 
 		immovable = true;
 
-		if (parameters != null)
-		{
+		if (parameters != null) {
+			// this gives the obstacle its init values
 			init(parameters);
 		}
 	}
 
-	public function init(parameters:ObstacleParameters)
-	{
+	public function init(parameters:ObstacleParameters) {
+		// revive the obstace to be drawn again
 		super.reset(parameters.x, parameters.y);
 
+		// set the velocity to whatever the speed is but neg
+		var upOrDown = FlxG.random.int(0, 1);
 		velocity.x = -parameters.speed;
 
+		if (upOrDown == 1) {
+			velocity.y = -(FlxG.random.float(parameters.speed / 2, parameters.speed));
+		} else {
+			velocity.y = (FlxG.random.float(parameters.speed / 2, parameters.speed));
+		}
```

Comment:
If we make these vertically-moving deadly obstacles a new subclass, then we don't have to worry about changing the Obstacle's base class behavior (and we can clean up that line in DeadlyObstacle that sets y velocity to 0).

Setting up code in the base class that subclasses have to undo is kinda inverted from the pattern we want to follow.